Backpack/Price/Classes/BPKPrice.swift

```diff
@@ -0,0 +1,203 @@
+        [
+            .foregroundColor: BPKColor.textSecondaryColor,
+            .font: BPKFont.makeFont(fontStyle: footnoteOrCaptionFontStyle()),
+            .strikethroughStyle: NSUnderlineStyle.single.rawValue,
+            .strikethroughColor: BPKColor.textSecondaryColor
+        ]
+    }()
+    
+    private lazy var priceLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: size == .large ? .textHeading2 : .textHeading4)
+        return label
+    }()
+    
+    private lazy var trailingTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var lineThroughTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var separatorLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var leadingTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var topTextStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [lineThroughTextLabel, separatorLabel, leadingTextLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingSm
+        return stackView
+    }()
+    
+    private lazy var priceStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [priceLabel, trailingTextLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingSm
+        stackView.alignment = .firstBaseline
+        return stackView
+    }()
+    
+    private lazy var containerStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [topTextStackView, priceStackView])
+        stackView.axis = .vertical
+        return stackView
+    }()
+    
+    public init(
+        price: String,
+        leadingText: String? = nil,
+        lineThroughText: String? = nil,
+        trailingText: String? = nil,
+        alignment: Alignment = .left,
+        size: Size = .large
+    ) {
+        self.price = price
+        self.leadingText = leadingText
+        self.lineThroughText = lineThroughText
+        self.trailingText = trailingText
+        self.alignment = alignment
+        self.size = size
+        
+        super.init(frame: .zero)
+        
+        leadingTextLabel.text = leadingText
+        separatorLabel.text = "•"
+        priceLabel.text = price
+        trailingTextLabel.text = trailingText
+        
+        lineThroughTextLabel.text = lineThroughText
+        applyLineThroughStyling()
```

Comment:
this seems to be part of the setup of the view, can we move this to the `setupView` method?